File: leo2-ui/src/main/java/org/deku/leo2/fx/modules/DepotMaintenanceController.java
package org.deku.leo2.fx.modules;

import javafx.fxml.FXML;
import javafx.fxml.Initializable;
import javafx.scene.Node;
import org.deku.leo2.Main;
import org.deku.leo2.fx.ModuleController;
import org.deku.leo2.fx.components.DepotDetailsController;
import org.deku.leo2.fx.components.DepotListController;
import org.deku.leo2.rest.v1.entities.Depot;

import java.net.URL;
import java.util.ResourceBundle;

/**
 * Created by masc on 22.09.14.
 */
public class DepotMaintenanceController extends ModuleController implements Initializable, DepotListController.Listener {
    @FXML
    private Node mDepotList;
    @FXML
    private DepotListController mDepotListController;
    @FXML
    private Node mDepotDetails;
    @FXML
    private DepotDetailsController mDepotDetailsController;

    @Override
    public String getTitle() {
        return Main.instance().getLocalizedString("menu.depots");
    }

    @Override
    public void initialize(URL location, ResourceBundle resources) {
        mDepotListController.setListener(this);
    }

    @Override
    public void onActivation() {
        mDepotListController.activate();
    }

    @Override
    public void onDepotListItemSelected(Depot depot) {
        mDepotDetailsController.setDepot(depot);
    }

    @Override
    public void dispose() {
        super.dispose();
        mDepotListController.dispose();
    }
}

File: leoz-ui/src/main/java/org/deku/leoz/ui/fx/modules/DepotMaintenanceController.kt
package org.deku.leoz.ui.fx.modules

import com.github.salomonbrys.kodein.Kodein
import com.github.salomonbrys.kodein.conf.global
import com.github.salomonbrys.kodein.instance
import com.github.salomonbrys.kodein.lazy
import javafx.fxml.FXML
import javafx.fxml.Initializable
import javafx.scene.Node
import javafx.scene.image.Image
import org.deku.leoz.ui.event.Event
import org.deku.leoz.ui.fx.ModuleController
import org.deku.leoz.ui.fx.components.DepotDetailsController
import org.deku.leoz.ui.fx.components.DepotListController
import io.reactivex.subjects.PublishSubject
import java.net.URL
import java.util.*

/**
 * Created by masc on 22.09.14.
 */
class DepotMaintenanceController : ModuleController(), Initializable {
    @FXML
    private lateinit var fxDepotList: Node
    @FXML
    private lateinit var fxDepotListController: DepotListController
    @FXML
    private lateinit var fxDepotDetails: Node
    @FXML
    private lateinit var fxDepotDetailsController: DepotDetailsController

    private val i18n: org.deku.leoz.ui.Localization by Kodein.global.lazy.instance()

    override val title: String
        get() = this.i18n.resources.getString("menu.depots")

    override val titleImage: Image by lazy { Image(this.javaClass.getResourceAsStream("/images/depot-144px.png")) }

    override val ovBusy: PublishSubject<Event<Boolean>> by lazy {
        fxDepotListController.ovBusy
    }

    override fun initialize(location: URL, resources: ResourceBundle) {
        fxDepotListController.ovItemSelected.subscribe {
            fxDepotDetailsController.station = it
        }
    }

    public override fun onActivation() {
        fxDepotListController.activate()
    }

    fun selectDepot(id: Int?) {
        fxDepotListController.requestDepotSelection(id)
    }

    override fun close() {
        super.close()
        fxDepotListController.close()
    }
}
